apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,453 @@
+        public void CompareTo_Equal(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(0, result);
+        }
+
+        [Theory]
+        [MemberData(nameof(NaNComparisionData))]
+        public void CompareTo_NaN(string input1, string input2, int expectedResult)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(expectedResult, result);
+        }
+
+        [Fact]
+        public void CompareTo_NegativeZero()
+        {
+            var regularZero = PgNumeric.Zero;
+            var negativeZero = PgNumeric.Parse("-0");
+            Assert.Equal(0, regularZero.CompareTo(negativeZero));
+            Assert.Equal(0, (IComparable)regularZero.CompareTo(negativeZero));
+            Assert.Equal(0, negativeZero.CompareTo(regularZero));
+            Assert.Equal(0, (IComparable)negativeZero.CompareTo(regularZero));
+        }
+
+        [Fact]
+        public void CompareTo_Null()
+        {
+            IComparable minValue = PgNumeric.MinValue;
+            Assert.Equal(1, minValue.CompareTo(null));
+        }
+
+        [Theory]
+        [InlineData("0.00", "001")]
+        [InlineData("0.00", "1")]
+        [InlineData("0.00", "0.01")]
+        [InlineData("-1.0", "1.000")]
+        [InlineData("-1.0", "0")]
+        [InlineData("-10.0", "0.00000009000000000")]
+        [InlineData("0", "1.00000000")]
+        [InlineData("1", "1.01")]
+        [InlineData("0.00000009", "0.0000008")]
+        [InlineData("1.999999999999999999999999", "2")]
+        [InlineData("99.90", "99.99")]
+        [InlineData("-1.000", "1.0")]
+        [InlineData("-123.0", "-1")]
+        [InlineData("-0.0009", "0")]
+        [InlineData("0.00000009", "0.00009")]
+        public void CompareTo_LessThan(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.True(result < 0);
+        }
+
+        [Theory]
+        [InlineData("001", "0.00")]
+        [InlineData("1", "0.00")]
+        [InlineData("0.01", "0.00")]
+        [InlineData("1.000", "-1.0")]
+        [InlineData("0", "-1.0")]
+        [InlineData("0.00000009000000000", "-10.0")]
+        [InlineData("1.00000000", "0")]
+        [InlineData("1.01", "1")]
+        [InlineData("0.0000008", "0.00000009")]
+        [InlineData("2", "1.999999999999999999999999")]
+        [InlineData("99.99", "99.90")]
+        [InlineData("1.0", "-1.000")]
+        [InlineData("-1", "-123.0")]
+        [InlineData("0", "-0.0009")]
+        [InlineData("0.00009", "0.00000009")]
+        public void CompareTo_GreaterThan(string input1, string input2)
```

Comment:
All the scenarios mentioned above are covered in existing test cases except for explicit `<` , `>` tests as PgNumeric doesn't implement `<`  and `>` operators but they are covered via `CompareTo`. Please advise if individual tests aren't fine. I believe it would be easier to test and isolate issues if we have specific tests for equality, less than and greater than comparison. Please let me know you thoughts and I will do as you suggest.

Reply:
It doesn't need to be done before we merge, but I think it's simpler to have *either* a single test that covers all non-equal comparisons, *or* multiple test methods but with a single set of test data.

Currently if we found some corner case, we'd need to add that test data in multiple places, and that's not great.

While I'm aware that some testing philosophies suggest a single assertion per test, my experience is that it's far more pragmatic to test multiple very-closely-related operations (such as "all comparisons") in a single test. We're unlikely to have a problem in greater-than but not in less-than, for example. (I can see inequality being potentially separated out.)